control-plane/p4RuntimeSerializer.cpp

```diff
@@ -1611,9 +1687,12 @@ P4RuntimeSerializer::generateP4Runtime(const IR::P4Program* program, cstring arc
     auto* p4RuntimeProgram = program->apply(p4RuntimeFixups);
     auto* evaluatedProgram = evaluator->getToplevelBlock();
 
-    BUG_CHECK(p4RuntimeProgram && evaluatedProgram,
-              "Failed to transform the program into a "
-              "P4Runtime-compatible form");
+    if (!p4RuntimeProgram || !evaluatedProgram) {
+        ::error(ErrorType::LEGACY_ERROR,
+                "Failed to transform the program into a "
```

Comment:
This message is a bit confusing. How about "error while generating P4Runtime API?"